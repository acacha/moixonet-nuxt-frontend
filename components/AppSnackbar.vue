<template>
  <v-snackbar
    v-model="show"
    :color="color"
  >
    {{ message }}
    <v-btn
      color="primary"
      text
      data-test="close_button"
      @click="toggle"
    >
      Tancar
    </v-btn>
  </v-snackbar>
</template>

<script>
export default {
  name: 'AppSnackbar',
  model: {
    prop: 'show',
    event: 'toggle'
  },
  props: {
    show: {
      type: Boolean,
      default: false
    },
    color: {
      type: String,
      validator: (value) => {
        // El valor debe coincidir con una de estas cadenas de texto
        return ['success', 'warning', 'error'].includes(value)
      }
    },
    message: {
      type: String,
      default: 'Put your message here'
    }
  },
  methods: {
    toggle () {
      this.$emit('toggle')
    }
  }
}
</script>
