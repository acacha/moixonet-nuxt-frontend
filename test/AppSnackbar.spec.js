import { shallowMount, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import Vue from 'vue'
import AppSnackbar from '../components/AppSnackbar'

Vue.use(Vuetify)

describe('AppSnackbar', () => {
  let opts

  const build = (options) => {
    options = options || opts
    const wrapperShallow = shallowMount(AppSnackbar, options)
    const wrapper = mount(AppSnackbar, options)

    return {
      wrapperShallow,
      wrapper,
      closeButton: () => wrapper.find('[data-test=close_button]')
    }
  }

  beforeEach(() => {
    opts = {}
  })

  test('is a Vue instance', () => {
    const { wrapperShallow } = build()
    expect(wrapperShallow.isVueInstance()).toBeTruthy()
  })

  test('renders the component with default props', () => {
    const { wrapperShallow } = build()
    expect(wrapperShallow.element).toMatchSnapshot()
  })

  test('renders the default message', () => {
    const { wrapper } = build({
      propsData: {
        'show': true
      }
    })
    expect(wrapper.text()).toMatch('Put your message here')
    expect(wrapper.element).toMatchSnapshot()
  })

  test('renders the passed message', () => {
    const { wrapper } = build({
      propsData: {
        'show': true,
        'message': 'Unexpected error'
      }
    })
    expect(wrapper.text()).toMatch('Unexpected error')
    expect(wrapper.element).toMatchSnapshot()
  })

  test('color validation', () => {
    const { wrapper } = build()
    const color = wrapper.vm.$options.props.color

    expect(color.type).toBe(String)
    expect(color.validator && color.validator('unvalid')).toBeFalsy()
    expect(color.validator && color.validator('error')).toBeTruthy()
  })

  test('renders the component', () => {
    const { wrapperShallow } = build({
      propsData: {
        'show': true
      }
    })
    expect(wrapperShallow.element).toMatchSnapshot()
  })

  test('can be closed', () => {
    const { closeButton, wrapper } = build({
      propsData: {
        'show': true
      }
    })
    closeButton().trigger('click')
    expect(wrapper.emitted().toggle).toBeTruthy()
    expect(wrapper.emitted().toggle.length).toBe(1)
  })
})
