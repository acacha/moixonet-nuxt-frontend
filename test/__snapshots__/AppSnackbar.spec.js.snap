// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSnackbar renders the component 1`] = `
<v-snackbar-stub
  color="success"
  timeout="6000"
  value="true"
>
  
  Put your message here
  
  <v-btn-stub
    activeclass=""
    color="white"
    data-test="close_button"
    tag="button"
    text="true"
    type="button"
  >
    
    Tancar
  
  </v-btn-stub>
</v-snackbar-stub>
`;

exports[`AppSnackbar renders the component with default props 1`] = `
<v-snackbar-stub
  color="success"
  timeout="6000"
>
  
  Put your message here
  
  <v-btn-stub
    activeclass=""
    color="white"
    data-test="close_button"
    tag="button"
    text="true"
    type="button"
  >
    
    Tancar
  
  </v-btn-stub>
</v-snackbar-stub>
`;

exports[`AppSnackbar renders the default message 1`] = `
<div
  class="v-snack v-snack--active v-snack--bottom"
  name="v-snack-transition"
>
  <div
    class="v-snack__wrapper success"
  >
    <div
      class="v-snack__content"
    >
      
  Put your message here
  
      <button
        class="v-btn v-btn--flat v-btn--text theme--light v-size--default white--text"
        data-test="close_button"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          
    Tancar
  
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`AppSnackbar renders the passed message 1`] = `
<div
  class="v-snack v-snack--active v-snack--bottom"
  name="v-snack-transition"
>
  <div
    class="v-snack__wrapper success"
  >
    <div
      class="v-snack__content"
    >
      
  Unexpected error
  
      <button
        class="v-btn v-btn--flat v-btn--text theme--light v-size--default white--text"
        data-test="close_button"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          
    Tancar
  
        </span>
      </button>
    </div>
  </div>
</div>
`;
